src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs

```diff
@@ -299,12 +326,26 @@ private async Task PreviewAndExecuteUpdateActionsForSinglePackage()
                     new GatherCache(),
                     sourceCacheContext);
 
+                // PackageReference projects don't support `Update-Package -Reinstall`. 
+                var relevantProjects = Projects;
+                if (Reinstall.IsPresent)
+                {
+                    var groupedProjects = Projects.GroupBy(e => e is BuildIntegratedNuGetProject);
+                    var buildIntegratedProjects = groupedProjects.Where(e => e.Key).FirstOrDefault();
+                    if (buildIntegratedProjects != null && buildIntegratedProjects.Any())
+                    {
+                        WarnForReinstallOfBuildIntegratedProjects(buildIntegratedProjects.AsEnumerable().Cast<BuildIntegratedNuGetProject>());
+                        var nonBuildIntegratedProjects = groupedProjects.Where(e => !e.Key).FirstOrDefault();
+                        relevantProjects = nonBuildIntegratedProjects?.ToList() ?? new List<NuGetProject>();
+                    }
+                }
```

Comment:
Could we have a private method that simply separates the PackageReference projects from Packages.config projects? Then the logic to warn on one set while use the other set becomes simpler.

Reply:
Yep, I can do that. 
Now that I have concept approval, I'll refactor that + try to have 1 method for warning as @erdembayar suggested.